src/publisher/index.ts

```diff
@@ -136,7 +140,29 @@ export class Publisher {
    * @param {PublishCallback} [callback] Callback function.
    */
   publishMessage(message: PubsubMessage, callback: PublishCallback): void {
+    // Construct publisher span and set context as message attribute
     const {data, attributes = {}} = message;
+    const spanAttributes = {
```

Comment:
If we made telemetry configurable, then this logic could be pulled into a conditional block.

Either way I would consider pulling this into a private helper.